src/jesse_validator_draft4.erl

```diff
@@ -1379,3 +1379,14 @@ maybe_external_check_value(Value, State) ->
     Fun ->
       Fun(Value, State)
   end.
+
+maybe_external_check_format(Value, Format, State) ->
```

Comment:
this function is the same as one defined in draft3. Could you consider moving it in some common place (eg, jesse_lib)? But it's debatable, because AFAIK, draft3 and draft4 have quite a lot of duplicate code already.

Reply:
just same place as 'maybe_external_check_value' and i thinking about mutating validators in future (when you validate AND transform data at once, so don't need to parse it twice)